### DISCUSSION TITLE
Enabling femu_debug option while running femu

### DISCUSSION THREAD

**Discussion by aadit0402** [MoatLab/FEMU#133] [General]
Hi Developers,

I have gone through the wiki provided on github page but could not find anything on "How to enable femu_debug" option in order to make some basic understanding on working mechanism of FEMU while running femu with different variants of SSDs. Would you like to help on this?

Thanks in advance,
Aditya

**Comment by aadit0402** [comment]
I see these are some comment added by default with github repo and wanted to see those in some trace file, how to enable these:

femu_debug("Starting FEMU in Blackbox-SSD mode ...\n");
hw/femu/femu.c:16:        femu_debug("shutting down NVMe Controller ...\n");
hw/femu/femu.c:18:        femu_debug("disabling NVMe Controller ...\n");
hw/femu/femu.c:22:        femu_debug("%s,clear_guest_notifier\n", __func__);
hw/femu/femu.c:576:    femu_debug("Destroying NVMe poller !!\n");
hw/femu/femu.c:595:    femu_debug("femu_exit starting!\n");
hw/femu/nvme-admin.c:280:        femu_debug("femu-nvme-poller [%d] created ...\n", i - 1);

any way to debug it with GDB? I would appreciate your response. 

Thanks,
Aditya

**Reply by huaicheng** [reply]
QEMU is just a normal program, so you can debug it using GDB just like other programs. If you want to dynamically toggle femu_debug(), a possible way is to define a specialized nvme command to trigger a global variable to control the invocations of femu_debug() calls.
